gtsfm/common/view_frustum.py

```diff
@@ -0,0 +1,146 @@
+        self.img_h_ = img_h
+        self.frustum_ray_len_ = frustum_ray_len
+
+    def get_frustum_vertices(self) -> Tuple[np.ndarray,np.ndarray,np.ndarray,np.ndarray,np.ndarray]:
+        """Obtain 3d positions of all 5 frustum vertices
+
+          (x,y,z)               (x,y,z)                (x,y,z)              (x,y,z)
+              \\=================//                      \\                   //
+               \\               //                        \\ 1-------------2 //
+        (-w/2,-h/2,fx)       (w/2,-h/2,fx)                 \\| IMAGE PLANE |//
+                 1-------------2                             |             |/
+                 |\\         //| IMAGE PLANE  (-w/2, h/2,fx) 3-------------4 (w/2, h/2,fx)
+                 | \\       // | IMAGE PLANE                  \\         //
+                 3--\\-----//--4                               \\       //
+                     \\   //                                    \\     //
+                      \\ //                                      \\   //
+                        O PINHOLE                                 \\ //
+                                                                    O PINHOLE
+        """
+        uv = np.array(
+            [
+                [self.img_w_ // 2, self.img_h_ // 2],  # optical center
+                [0, 0],  # top-left
+                [self.img_w_ - 1, 0],  # top-right
+                [0, self.img_h_ - 1],  # bottom-left
+                [self.img_w_ - 1, self.img_h_ - 1],  # bottom-right
+            ]
+        )
+
+        ray_dirs = compute_pixel_ray_directions_vectorized(
+            uv, self.fx_, self.img_w_, self.img_h_
+        )
+        v0 = ray_dirs[0] * 0
+        v1 = ray_dirs[1] * self.frustum_ray_len_
+        v2 = ray_dirs[2] * self.frustum_ray_len_
+        v3 = ray_dirs[3] * self.frustum_ray_len_
+        v4 = ray_dirs[4] * self.frustum_ray_len_
+
+        return v0, v1, v2, v3, v4
+
+    def get_mesh_edges_camframe(self) -> np.ndarray:
+        """Return 8 edges defining the frustum mesh, in camera coordinate frame.
+
+           Edge ordering below:
+
+                0
+            .-------.
+           3|\\1   /|4
+            ._\\ 2/_.
+            \\ \\/ /
+             5\\../6
+
+           Returns:
+               edges_camfr: array of shape (8,2,2) representing 8 polylines in camera frame
+        """
+        v0, v1, v2, v3, v4 = self.get_frustum_vertices()
+
+        e0 = np.array([v1, v2]) # top-left to top-right (e0)
+        e1 = np.array([v0, v1]) # optical center to top-left (e1)
+        e2 = np.array([v0, v2]) # optical center to top-right (e2)
+        e3 = np.array([v3, v1]) # bottom-left to top-left (e3)
+        e4 = np.array([v4, v2]) # bottom-right to top-right (e4)
+        e5 = np.array([v0, v3]) # optical center to bottom-left (e5)
+        e6 = np.array([v0, v4]) # optical center to bottom-right (e6)
+        e7 = np.array([v3, v4]) # bottom-left to bottom-right (e7)
```

Comment:
I think using a for loop to connect the origin to the 4 points might be cleaner. 
And another for loop could be used to connect v1 - v2, v2 - v3, v3 - v4, v4- v1. 
Also think it might be better to move this to a different helper and use it in both get_mesh_edges_camframe and get_mesh_edges_worldframe.